nexus/db-queries/src/db/datastore/rack.rs

```diff
@@ -214,6 +215,63 @@ impl DataStore {
         Ok(())
     }
 
+    // Return the subnet for the rack
+    pub async fn rack_subnet(
+        &self,
+        opctx: &OpContext,
+        rack_id: Uuid,
+    ) -> Result<IpNetwork, Error> {
+        opctx.authorize(authz::Action::Read, &authz::FLEET).await?;
+        let conn = self.pool_connection_authorized(opctx).await?;
+        use db::schema::rack::dsl;
+        // It's safe to unwrap the returned `rack_subnet` because
+        // we filter on `rack_subnet.is_not_null()`
+        dsl::rack
+            .filter(dsl::id.eq(rack_id))
+            .filter(dsl::rack_subnet.is_not_null())
+            .select(dsl::rack_subnet)
+            .limit(1)
+            .first_async::<Option<IpNetwork>>(&*conn)
```

Comment:
You could use `.get_result_async` here instead of `.first_async` here, if you want to throw a "Not Found" error.

See: https://docs.rs/async-bb8-diesel/0.1.0/async_bb8_diesel/trait.AsyncRunQueryDsl.html

These are all the `..._async` versions of: https://docs.rs/diesel/latest/diesel/query_dsl/trait.RunQueryDsl.html

From `get_result` specifically:


> Err(NotFound) will be returned if the query affected 0 rows.
> You can call .optional() on the result of this if the command was optional to get back a Result<Option<U>>

Reply:
(this might be applicable for other spots you're using `.first_async` ?)

Reply:
This is a great tip! Thank you! I had incorrectly assumed that `first_async` would also return a "Not Found" error. I will fix and use thisl.

Reply:
Hmm, So it looks like first_async does the same thing:
> Returns Ok(record) if found, and Err(NotFound) if no results are returned. If the query truly is optional, you can call .optional() on the result of this to get a Result<Option<U>>.

The problem requiring the `Option` is the nullable field.